luigi/contrib/s3.py

```diff
@@ -504,21 +504,23 @@ def list(self, path, start_time=None, end_time=None, return_key=False):  # backw
 
     @staticmethod
     def _get_s3_config(key=None):
-        defaults = dict(configuration.get_config().defaults())
-        try:
-            config = dict(configuration.get_config().items('s3'))
-        except (NoSectionError, KeyError):
-            return {}
-        # So what ports etc can be read without us having to specify all dtypes
-        for k, v in six.iteritems(config):
+        if os.environ.get("LUIGI_CONFIG_PARSER") != "toml":
```

Comment:
If I remain correctly, the config type can be specified not only from environment variables, but also through passing from the code. More stable way is check the instance of the config: `if isinstance(configuration.get_config(), ...)`